Use base configs for molecule setup (#38)
UCL-MIRSG/ansible-collection-infra: roles/README.md

This collection is tested using Ansible Molecule.

Each role has its own Molecule configuration, which can be found it the `molecule/`
folder of each role. Molecule base configurations are used to reduce to amount
of duplication in the setup for testing each role. There are two base configurations
that correspond to two [Molecule
scenarios](https://ansible.readthedocs.io/projects/molecule/getting-started/#molecule-scenarios)

> one for testing on CentOS 7 and another for testing on Rocky 9. The base configurations

Reviewer: ```suggestion
one for testing on CentOS 7 and another for testing on RockyLinux 9. The base configurations
```

are in the `ansible_collections/mirsg/infrastructure/tests` folder.

Playbooks are tested at the collection level, i.e. their configuration is in the
top-level `tests` folder for the collection. For each playbook, there are two
Molecule scenerios - one for CentOS 7 and one for Rocky 9, e.g.

- `ansible_collections/mirsg/infrastructure/tests/molecule/centos7_monitoring`

> - `ansible_collections/mirsg/infrastructure/tests/molecule/rocky9_monitoring`

Reviewer: Do we need the note about testing playbooks here?

Author: oops no, probably not!

### Add a molecule configuration

Create a `molecule` folder in your role:

```bash
mkdir my_roles/molecule
```

Add the `converge.yml` playbook to the `molecule/` folder,
and create folders for the `centos7` and `rocky9` scenarios. See
[`mirsg.infrastructure.firewalld` for an example](./firewalld/).

> If necessary, add [inventory `group_vars`](./firewalld/molecule/resources/inventory/group_vars/)

Reviewer: ```suggestion
If necessary, add inventory `group_vars` under a `resources` subfolder: `my-role/molecule/resources/inventory/group_vars`
```

for your role.

> You may also need to add [`prepare`](./firewalld/molecule/resources/prepare.yml) or

Reviewer: ```suggestion
You may also need to add a `prepare` or
```

> [`verify`](./firewalld/molecule/resources/prepare.yml) steps for your role.

Reviewer: ```suggestion
`verify` playbooks for your role. These can be added to the `resources` subfolder if you have multiple scenarios that will share the playbooks, or in the default scenario folder.
```

>   molecule-firewalld:

Reviewer: ```suggestion
  molecule-my_role:
```
